libmariadb/my_stmt.c

```diff
@@ -299,10 +321,62 @@ static int stmt_cursor_fetch(MYSQL_STMT *stmt, uchar **row)
 
 void mthd_stmt_flush_unbuffered(MYSQL_STMT *stmt)
 {
+  MYSQL *mysql = stmt->mysql;
```

Comment:
This whole logic has been borrowed from the updated code in mariadb-3x PR [here](https://github.com/mariadb-corporation/mariadb-connector-c/pull/131/files#diff-c5b3f02f161404ffabbd6f5e293e1391R370)